flink-cyber/metron-parser-chain/parser-chains-config-service/frontend/parser-chains-client/src/app/chain-page/components/live-view/sample-data-form/sample-data-text-folder-input/sample-data-text-folder-input.component.ts

```diff
@@ -0,0 +1,161 @@
+import {Component, EventEmitter, Input, OnInit, Output} from '@angular/core';
+import {
+    SampleDataInternalModel,
+    SampleDataModel,
+    SampleDataType,
+    SampleTestStatus
+} from "../../models/sample-data.model";
+import {EntryParsingResultModel} from "../../models/live-view.model";
+import {select, Store} from "@ngrx/store";
+import {SampleDataTextFolderInputState} from "./sample-data-text-folder-input.reducers";
+import {
+    getEditModalVisible,
+    getExecutionStatus,
+    getRunResults,
+    getSampleData, getSampleFolderPath
+} from "./sample-data-text-folder-input.selectors";
+import {Observable} from "rxjs";
+import {
+    FetchSampleListTriggeredAction, SampleFolderViewInitializedAction,
+    SaveSampleListTriggeredAction,
+    ShowEditModalAction
+} from "./sample-data-text-folder-input.actions";
+import {FormBuilder, FormGroup} from "@angular/forms";
+import {liveViewInitialized} from "../../live-view.actions";
+
+@Component({
+    selector: 'app-sample-data-text-folder-input',
+    templateUrl: './sample-data-text-folder-input.component.html',
+    styleUrls: ['./sample-data-text-folder-input.component.scss']
+})
+export class SampleDataTextFolderInputComponent implements OnInit {
+
+    @Input() chainConfig: {};
+    @Output() sampleDataChange = new EventEmitter<SampleDataModel>();
+    @Output() sampleDataForceChange = new EventEmitter<SampleDataInternalModel[]>();
+
+    isExecuting$: Observable<boolean>;
+    runResults$: Observable<Map<number, {
+        status: SampleTestStatus,
+        expected: string,
+        result: string,
+        failure: boolean,
+        raw: EntryParsingResultModel[]
+    }>>;
+    sampleData$: Observable<SampleDataInternalModel[]>;
+    sampleFolderPath$: Observable<string>;
+    editSampleModalVisible$: Observable<boolean>;
+
+    expandSet = new Set<number>();
+
+    folderForm!: FormGroup;
+    currentSampleData: SampleDataInternalModel[];
+
+    selectedSample: [number, SampleDataInternalModel] = [null, null];
```

Comment:
maybe it would be better to use map instead of an array? 
```suggestion
    selectedSample: [id:number]: SampleDataInternalModel;
```